cpp/tests/strings/integers_tests.cu

```diff
@@ -164,20 +164,38 @@ TEST_F(StringsConvertTest, HexToInteger)
     h_strings.end(),
     thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
 
-  std::vector<int32_t> h_expected;
-  for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
-    if (*itr == nullptr)
-      h_expected.push_back(0);
-    else
-      h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
+  {
+    std::vector<int32_t> h_expected;
+    for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
+      if (*itr == nullptr)
+        h_expected.push_back(0);
+      else
+        h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
```

Comment:
Why not use stoi, may be I am unaware of something .

Reply:
The `std::stoi` throws a C++ exception when values are outside it's range and I want to just truncate the value using a cast in this case.

Reply:
You can use a `std::stringstream` to `<<` the string into and then use `>>` to extract the int/long, etc:
```
#include <iostream>
#include <string>
#include <sstream>

template<typename T>
T from_string(std::string const& str)
{
    T result;
    std::stringstream ss;
    ss<<str;
    ss>>result;
    return result;
}

int main(void)
{
    auto num_str = "12345";
    std::string str_num{num_str};
    auto value = from_string<int>(str_num);
    std::cout<<"value: "<<value<<'\n';
}
```